core/src/main/java/org/bitcoinj/coinjoin/CoinJoinBroadcastTx.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright 2022 Dash Core Group
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.bitcoinj.coinjoin;
+
+import org.bitcoinj.core.MasternodeSignature;
+import org.bitcoinj.core.Message;
+import org.bitcoinj.core.NetworkParameters;
+import org.bitcoinj.core.ProtocolException;
+import org.bitcoinj.core.Sha256Hash;
+import org.bitcoinj.core.Transaction;
+import org.bitcoinj.core.TransactionOutPoint;
+import org.bitcoinj.core.UnsafeByteArrayOutputStream;
+import org.bitcoinj.core.Utils;
+import org.bitcoinj.crypto.BLSPublicKey;
+import org.bitcoinj.crypto.BLSSignature;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+
+// dstx
+public class CoinJoinBroadcastTx extends Message {
+    private static final Logger log = LoggerFactory.getLogger(CoinJoinQueue.class);
+
+    private Transaction tx;
+    private TransactionOutPoint masternodeOutpoint;
+    private MasternodeSignature signature;
+    private long signatureTime;
```

Comment:
this is another place where we will have to have Dash Core (perhaps v19) updated for this message.

`masternodeOutpoint` cannot be linked to a masternode (`SimplifiedMasternodeListEntry`), which is the class that DashJ uses to keep track of masternodes.